### PR TITLE
Upload to ghcr.io

### PR DESCRIPTION
**Pull request by hamishmack**
This should upload tagged docker images to

```
ghcr.io/input-output-hk/cardano-node:$GITHUB_REF_NAME
ghcr.io/input-output-hk/cardano-submit-api:$GITHUB_REF_NAME
```

### CODE REVIEW

**Comment by hamishmack** [comment]
In order to make sure the docker images can be fetched from `cache.iog.io` this PR also removes the setting of `getrev` to `"000...000" for CI builds of PRs.  This should be ok because the setting of git revision is done much later in the build now (instead of embedding it in the source code the binaries are updated with the `gitrev` after they are linked.

File: .github/workflows/release-ghcr.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  build:
+    name: "Upload to ghcr.io"
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install Nix with good defaults
+      uses: input-output-hk/install-nix-action@v20
+      with:
+        extra_nix_config: |
+          trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
+          substituters = https://cache.iog.io/ https://cache.nixos.org/
```
Comment (andreabedini) [inline .github/workflows/release-ghcr.yaml:21]: Maybe we can skip this entirely and just use `--accept-flake-config`

File: .github/workflows/release-ghcr.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        nix_path: nixpkgs=channel:nixos-unstable
```
Comment (andreabedini) [inline .github/workflows/release-ghcr.yaml:22]: This can also go.

**Comment by hamishmack** [comment]
Thanks heaps for fixing this up @andreabedini.  Changes look good.

**Review by andreabedini — APPROVED** [review]

**Pull request merged**